File: src/contests/cg/puzzle/objectivec/google-interview---the-two-egg-problem.m
// 🎮 CodinGame Puzzle - google-interview---the-two-egg-problem
// https://www.codingame.com/training/hard/google-interview---the-two-egg-problem

#import <Foundation/Foundation.h>

int main() {
    long long n;
    scanf("%lld", &n);
    long long k = 0;
    while (k * (k + 1) / 2 < n) k++;
    printf("%lld\n", k);
    return 0;
}

File: src/contests/cg/puzzle/php/google-interview---the-two-egg-problem.php
<?php
// 🎮 CodinGame Puzzle - google-interview---the-two-egg-problem
// https://www.codingame.com/training/hard/google-interview---the-two-egg-problem

$n = (int) trim(fgets(STDIN));
$k = 0;
while (intdiv($k * ($k + 1), 2) < $n) {
    $k++;
}
echo $k . "\n";
